addons/web_editor/static/src/scss/wysiwyg.scss

```diff
@@ -265,6 +265,7 @@ body .modal {
                 .o_existing_attachment_remove {
                     @include o-position-absolute($top: 0, $right: 0);
                     border-radius: 0 0 0 2px;
+                    z-index: 1;
```

Comment:
- You don't need a z-index here: modifiy the DOM so that the button is placed at the end of the container so that it naturally appears above.
- Mention the commit that broke the behavior when you can, this is that one in this case: https://github.com/odoo/odoo/commit/94b59b1972e1a18e18dfcd6c2ded495391380b9a